### DISCUSSION TITLE
Proper status code type?

### DISCUSSION THREAD

**Discussion by javierguzman** [fastify/fastify#4950] [Q&A] [answered]
Hello all,

I have created my custom error:

```
export class ApiError extends Error {
  statusCode: number;

  constructor(code: number, message: string) {
    super(message);
    this.statusCode = code;
  }
}
```

However, when I catch to make the response it complains because my statusCode is a number:
```
      } catch (error) {
        request.log.error(error);
        await trx.rollback();
        if (error instanceof ApiError) {
          return reply.status(error.statusCode).send({ error: error.message });
                                                    
        }
        return reply.code(500).send({ error: 'Something went wrong' });
      }
    }
```
How can I make this work? I have tried to use FastifyReply["code"], etc. but it did not work.

Thank you in advance and regards

**Comment by Uzlopak** [comment]
Why dont you utilize fastify-error?

To get some help, you should provide a repository with a minimal viable code example.

**Reply by javierguzman** [reply, marked as answer]
I did not know about fastify-error but I have managed to solve it, I placed my code in this discussion for future people https://github.com/fastify/fastify/discussions/4740
